rfd/0016-dynamic-configuration.md

```diff
@@ -231,29 +230,55 @@ This logic implies Choices 2.B and 3.B.
    immediately.
 
 2. If called without `--force` and the fetched resource does not have
-   `origin:defaults` then print
+   `origin: defaults` then print an error:
    ```
-   error: non-default cluster auth preference already exists, use -f or --force
-   flag to overwrite
+   non-default cluster auth preference already exists, use -f or --force flag
+   to overwrite
    ```
 
 3. If called with `--force` and the fetched resource does not have `origin:
    config-file` then store the resource specified in the provided YAML file
    in the backend.
 
 4. If called with `--force` and the fetched resource has `origin: config-file`
-   then print
+   then print an error:
    ```
-   error: This resource is managed by static configuration. We recommend
-   removing configuration from teleport.yaml, restarting the servers and trying
-   this command again.
+   This resource is managed by static configuration. We recommend removing
+   configuration from teleport.yaml, restarting the servers and trying this
+   command again.
 
    If you would still like to proceed, re-run the comand with both --force and
    --confirm flags.
    ```
 
-   This requires adding support for `--confirm` flag (that works only in
-   conjunction with `--force`).  Invocation with both `--force` and `--confirm`
-   always commits the YAML resource.
+   Invocation with both `--force` and `--confirm` will temporarily replace the
+   stored resource with the YAML one.
 
 This logic implies Choice 5.A and an adaptation of Choice 6.A.
+
+### `tctl rm`
+
+The `tctl rm` subcommand can be used to reset dynamic resources back to their defaults.
+
+1. Fetch the resource currently stored in the backend; on error return
+   immediately.
+
+2. If the fetched resource does not have `origin: config-file`, replace the
+   stored resource with the default one.
```

Comment:
It is generally assumed that to have a dynamic config in place there should be no related custom values in the config file. But you're right, there may be some cases (the aforementioned temporal overrides) that introduce dynamic configuration by overriding the present static configuration. It might make sense for `tctl rm` to remove such temporary experimental dynamic configs and bring back the overridden `config-file` values.

As we don't really keep the config file values around, implementing this would be pretty complicated, I think.

Reply:
Fair enough.